File: check_stats.py
import re
import json

with open('d:\\AntiGravity\\Rich3\\officers.js', 'r', encoding='utf-8') as f:
    text = f.read()

pattern = r'\{ id:\s*\d+,\s*name:\s*"([^"]+)",\s*faction:\s*(\d+),\s*stats:\s*\{\s*1:\s*(\d+),\s*2:\s*(\d+),\s*3:\s*(\d+),\s*4:\s*(\d+),\s*5:\s*(\d+),\s*6:\s*(\d+)\s*\}\s*\}'
matches = re.findall(pattern, text)

factions = {1: [], 2: [], 3: [], 4: []}

for match in matches:
    name = match[0]
    faction = int(match[1])
    total = sum(int(x) for x in match[2:])
    if faction in factions:
        factions[faction].append(total)

for i in range(1, 5):
    if factions[i]:
        total_sum = sum(factions[i])
        avg = total_sum / len(factions[i])
        print(f"Faction {i} ({len(factions[i])} officers): Total = {total_sum}, Average = {avg:.2f}")

File: check_stats.js
const fs = require('fs');

const code = fs.readFileSync('d:\\AntiGravity\\Rich3\\officers.js', 'utf8');

const regex = /\{ id: \d+, name: "([^"]+)", faction: (\d+), stats: \{ 1: (\d+), 2: (\d+), 3: (\d+), 4: (\d+), 5: (\d+), 6: (\d+) \} \}/g;

let match;
const factions = { 1: [], 2: [], 3: [], 4: [] };

while ((match = regex.exec(code)) !== null) {
    const faction = parseInt(match[2]);
    const total = parseInt(match[3]) + parseInt(match[4]) + parseInt(match[5]) + parseInt(match[6]) + parseInt(match[7]) + parseInt(match[8]);
    factions[faction].push(total);
}

for (let i = 1; i <= 4; i++) {
    const sum = factions[i].reduce((a, b) => a + b, 0);
    const avg = sum / factions[i].length;
    console.log(`Faction ${i} (${factions[i].length} officers): Total = ${sum}, Average = ${avg.toFixed(2)}`);
}
